mxcubeweb/core/components/samplechanger.py

```diff
@@ -413,181 +408,3 @@ def _gripper_changed(self):
-        sample_mount_device.get_loaded_sample()
-        and sample_mount_device.get_loaded_sample().get_address() == data_model.loc_str
-    ):
-        return
-
-    if hasattr(sample_mount_device, "__TYPE__"):
-        if sample_mount_device.__TYPE__ in ["Marvin", "CATS"]:
-            element = "%d:%02d" % loc
-            sample = {"location": element, "sampleID": element}
-            mxcube.sample_changer.mount_sample_clean_up(sample)
-        else:
-            sample = {
-                "location": data_model.loc_str,
-                "sampleID": data_model.loc_str,
-            }
-
-            # in this case the sample changer takes the sample from an Harvester
-            # We Harvest the sample and make it ready to load
-            if mount_from_harvester:
-                mxcube.harvester.queue_harvest_sample(data_model)
-
-            try:
-                res = mxcube.sample_changer.mount_sample_clean_up(sample)
-            except RuntimeError:
-                res = False
-            logging.getLogger("user_level_log").info(
-                "Sample loading res: %s" % str(res)
-            )
-
-            # We need to investigate if the comment below is still valid
-            if not res == False:  # noqa: E712
-                # WARNING: explicit test of False return value.
-                # This is to preserve backward compatibility (load_sample was
-                # supposed to return None); if sample could not be loaded, but
-                # no exception is raised, let's skip the sample
-
-                raise queue_entry.QueueSkipEntryException(
-                    "Sample changer could not load sample", ""
-                )
-
-    # Harvest Next sample while loading current
-    if mount_from_harvester and not HWR.beamline.harvester.get_room_temperature_mode():
-        mxcube.harvester.queue_harvest_next_sample(data_model)
-
-    robot_action_dict["endTime"] = time.strftime("%Y-%m-%d %H:%M:%S")
-    if sample_mount_device.has_loaded_sample():
-        robot_action_dict["status"] = "SUCCESS"
-    else:
-        robot_action_dict["message"] = "Sample was not loaded"
-        robot_action_dict["status"] = "ERROR"
-
-    HWR.beamline.lims.store_robot_action(robot_action_dict)
-
-    if not sample_mount_device.has_loaded_sample():
-        # Disables all related collections
-        logging.getLogger("user_level_log").info("Sample not loaded")
-        raise queue_entry.QueueSkipEntryException("Sample not loaded", "")
-    else:
-        signals.loaded_sample_changed(sample_mount_device.get_loaded_sample())
-        logging.getLogger("user_level_log").info("Sample loaded")
-        dm = HWR.beamline.diffractometer
-
-        if mount_from_harvester:
-            try:
-                logging.getLogger("user_level_log").info(
-                    "Start Auto Harvesting Centring"
-                )
-
-                computed_offset = (
-                    HWR.beamline.harvester.get_offsets_for_sample_centering()
```

Comment:
Here also,  to be ported here on this PR
https://github.com/mxcube/mxcubecore/pull/971